Sources/Core/Yoakke.Reporting/Present/TextDiagnosticsPresenter.cs

```diff
@@ -126,6 +148,11 @@ private void WriteSourceGroup(IEnumerable<SourceDiagnosticInfo> infos)
 
             // Generate all line primitives
             var linePrimitives = this.CollectLinePrimitives(infos).ToList();
+            // Trim empty source lines at edges
+            if (this.Style.TrimEmptySourceLinesAtEdges)
+            {
+                linePrimitives = TrimEmptySourceLinesAtEdges(linePrimitives).ToList();
```

Comment:
Is this the right place to do this? The thing is that we `...` out big gaps, don't we want those to participate in this "optimization"? For example, if we have
```
2 | hello
  | ^
3 | 
  | ...
7 | 
8 | bye
  | ^
```
We might want to just "merge in" lines 3 and 7 so we'd just have
```
2 | hello
  | ^ 
  | ...
8 | bye
  | ^
```
Also, a test for this case would be nice!

Reply:
Done! I have used this specific example as a test unit (`EmptyLineTrimmingWithDots`).